action.js

```diff
@@ -53,28 +58,36 @@ function initialTag(tag) {
 }
 
 async function existingTags() {
-  const options = octokit.git.listMatchingRefs.endpoint.merge({
-    ...requestOpts,
-    ref: 'tags',
-  })
-
-  core.info(
-    'Fetching existing tags. This may take a while on large repositories.'
-  )
+  core.info('Fetching tags.')
 
   return await octokit
-    .paginate(options)
-    .then((refs) => {
-      core.info('Tags fetch complete.')
-      return refs.reverse()
+    .graphql(
+      `{
+        repository(owner: "${owner}", name: "${repo}") {
+          refs(
+            first: 100
```

Comment:
can we make this a configurable input and default it to something small like 5 or 10?